crates/core/src/surfnet/svm.rs

```diff
@@ -937,91 +944,177 @@ impl SurfnetSvm {
-            transactions.push(transaction);
-        }
+                            VersionedMessage::V0(v0_msg) => (
+                                v0_msg.header,
+                                v0_msg.account_keys.iter().map(|k| k.to_string()).collect(),
+                                v0_msg
+                                    .instructions
+                                    .iter()
+                                    .map(|ix| UiCompiledInstruction::from(ix, None))
+                                    .collect(),
+                                Some(
+                                    v0_msg
+                                        .address_table_lookups
+                                        .iter()
+                                        .map(UiAddressTableLookup::from)
+                                        .collect(),
+                                ),
+                            ),
+                        };
+
+                    let inner_instructions = OptionSerializer::Some(
+                        meta.inner_instructions
+                            .iter()
+                            .enumerate()
+                            .map(|(i, inner_ixs)| UiInnerInstructions {
+                                index: i as u8,
+                                instructions: inner_ixs
+                                    .iter()
+                                    .map(|ix| {
+                                        UiInstruction::Compiled(UiCompiledInstruction::from(
+                                            &ix.instruction,
+                                            Some(ix.stack_height as u32),
+                                        ))
+                                    })
+                                    .collect(),
+                            })
+                            .collect(),
+                    );
+
+                    let transaction = EncodedTransactionWithStatusMeta {
+                        transaction: EncodedTransaction::Json(UiTransaction {
+                            signatures: tx.signatures.iter().map(|s| s.to_string()).collect(),
+                            message: UiMessage::Raw(UiRawMessage {
+                                header,
+                                account_keys,
+                                recent_blockhash: tx.get_recent_blockhash().to_string(),
+                                instructions,
+                                address_table_lookups,
+                            }),
+                        }),
+                        meta: Some(UiTransactionStatusMeta {
+                            err: err.clone(),
+                            status: err.map(|e| Err(e)).unwrap_or(Ok(())),
+                            inner_instructions,
+                            log_messages: OptionSerializer::Some(meta.logs.clone()),
+                            return_data: OptionSerializer::None,
```

Comment:
The `meta` field has `return_data` that you should be converting and returning here